client-side/src/popup/main.js

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { usePopupManager } from "react-popup-manager";
+import { Popup } from "./popup";
+
+export const Main = () => {
+  const popupManager = usePopupManager();
+  const openPopup = () => {
+    // open popup with it's needed `props` and an `onClose` callback function
+    popupManager.open(Popup, {
+      popupContent: '',
+      onClose: (...params) => console.log('modal has closed with:', ...params), // modal has closed with: param param2 param3
+    }); 
+  }
+  return (
+      <div>
+        <button onClick={() => openPopup()}>
```

Comment:
I can't understand what does this Main component used for, and why do you need to render it in App. I'd be happy to get explanation in Slack.

Reply:
i changed